src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,218 @@
+    end
+    push!(relations, poly-vars[2*s+1])
+
+    # h_F = h_E - sum x_G where G is a proper flat containing F
+    for i in 1:s
+        F = proper_flats[i]
+        poly = ring(0)
+        for index in findall(issubset(F,G) for G in proper_flats)
+            poly+= vars[index]
+        end
+        push!(relations, poly+vars[s+i]-vars[2*s+1]) #add h_F and subtract h_E
+    end
+    return relations
+end
+
+@doc Markdown.doc"""
+    augmented_chow_ring(M::Matroid)
+
+Return an augmented Chow ring of a matroid. As described in [BHMPW20](@cite). 
+
+Implemented by Fedor Glazov
+
+# Example
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = augmented_chow_ring(M);
+```
+"""
+function augmented_chow_ring(M::Matroid)
+    Flats = flats(M)
+    sizeFlats = length(Flats)
+    n = size_groundset(M)
+
+    is_loopless(M) || error("Matroid has loops")
+    sizeFlats>3 || error("Matroid has too few flats")
+
+    proper_flats = Flats[1:sizeFlats-1]
+    element_var_names = [string("y_", S) for S in M.groundset]
+
+    flat_var_names = ["x_{" * join(S, ",") * "}" for S in proper_flats]
+    var_names = vcat(element_var_names, flat_var_names)
+    s = length(var_names)
+
+    ring, vars = PolynomialRing(QQ, var_names)
+    element_vars = vars[1:n]
+    flat_vars = vars[n+1:s]
+
+    I = augmented_linear_relations(ring, proper_flats, element_vars, flat_vars, M)
+    J = augmented_quadratic_relations(ring, proper_flats, element_vars, flat_vars, M)
+
+    chow_modulus = ideal(ring, vcat(I, J))
+    chow_ring, projection = quo(ring, chow_modulus)
+
+    return chow_ring
+end
+
+function augmented_linear_relations(ring::MPolyRing, proper_flats::GroundsetType, element_vars::Vector, flat_vars::Vector, M::Matroid)
+    n = size_groundset(M)
+    relations = Vector{elem_type(ring)}(undef,n)
+    i = 1
+    for element in M.groundset
+        relations[i] = element_vars[i]
+        j = 1
+        for proper_flat in proper_flats
+            if !(element in proper_flat)
+                relations[i] -= flat_vars[j]
+            end
+            j += 1
+        end
+        i += 1
+    end
+    return relations
+end
+
+function augmented_quadratic_relations(ring::MPolyRing, proper_flats::Vector, element_vars::Vector, flat_vars::Vector, M::Matroid)
+    incomparable_polynomials = quadratic_relations(ring, proper_flats, flat_vars)
+    xy_polynomials = elem_type(ring)[]
+
+    i = 1
+    for element in M.groundset
+        j = 1
+        for proper_flat in proper_flats
+            if !(element in proper_flat)
+                push!(xy_polynomials, element_vars[i] * flat_vars[j])
+            end
+            j += 1
+        end
+        i += 1
+    end
+
+    return vcat(incomparable_polynomials, xy_polynomials)
+end
+
+"""
+A helper function to select indicies of a vector that do `include` elments of a given set and `exclude` anothers
+"""
+function select(include::Union{AbstractVector,Set},exclude::Union{AbstractVector,Set},set::Union{AbstractVector,Set})
```

Comment:
This is a very specific function with a very generic name. As this is a helper function, it should probably be renamed to `_select`.

Reply:
Done. But is there somewhere else a function like that?

Reply:
No, not that I know of. But there are finitely many good function names and reserving `select` for a helper function might come back to bite us in the future.

Reply:
I do understand that, but thought this kind of function would exists anyways

Reply:
It seems like a fairly special operation, and "fundamental" to building other stuff, so I'd be surprised if it existed elsewhere on Julia or Oscae